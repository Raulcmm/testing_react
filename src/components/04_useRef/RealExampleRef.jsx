import React, { useState } from "react";

import MultipleCustomHooks from "../03_examples/MultipleCustomHooks.jsx";
import "../02_useeffect/useeffect.css";
const RealExampleRef = () => {
	const [show, setShow] = useState(false);
	return (
		<div>
			<h1>Real example ref</h1>
			<hr />
			{show && <MultipleCustomHooks />}

			<button className="btn btn-primary" onClick={() => setShow(!show)}>
				Show/hide
			</button>
		</div>
	);
};

export default RealExampleRef;
