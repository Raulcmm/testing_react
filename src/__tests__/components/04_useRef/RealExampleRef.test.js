import React from "react";
import { shallow } from "enzyme";
import RealExampleRef from "../../../components/04_useRef/RealExampleRef";

describe("Pruebas en RealExampleRef", () => {
	test("Debe mostrarse correctamente", () => {
		const wrapper = shallow(<RealExampleRef />);
		expect(wrapper).toMatchSnapshot();
	});

	test("Debe mostrarse el componente MultipleCustomHooks", () => {
		const wrapper = shallow(<RealExampleRef />);
		wrapper.find("button").simulate("click");

		expect(wrapper.find("MultipleCustomHooks").exists()).toBeTruthy();
	});
});
