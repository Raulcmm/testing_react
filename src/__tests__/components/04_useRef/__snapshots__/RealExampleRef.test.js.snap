// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en RealExampleRef Debe mostrarse correctamente 1`] = `
<div>
  <h1>
    Real example ref
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Show/hide
  </button>
</div>
`;
